src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -371,6 +371,56 @@ class SumArrayAggregator[T](implicit ev: scala.math.Numeric[T], ct: ClassTag[T])
   def copy() = new SumArrayAggregator()
 }
 
+class MaxAggregator[T, BoxedT >: Null](implicit ev: NumericPair[T, BoxedT], ct: ClassTag[T]) extends TypedAggregator[BoxedT] {
+
+  import ev.numeric
+  import Ordering.Implicits._
+
+  var _state: T = ev.numeric.zero
+  var _isNull: Boolean = true
+
+  def result: BoxedT = if (_isNull) null else ev.box(_state)
+
+  def seqOp(x: Any) {
+    if (x != null && (_isNull || ev.unbox(x.asInstanceOf[BoxedT]) > _state)) {
+      _isNull = false
+      _state = ev.unbox(x.asInstanceOf[BoxedT])
+    }
+  }
+
+  def combOp(agg2: this.type) {
+    if (!agg2._isNull && (_isNull || agg2._state > _state))
+      _state = agg2._state
+  }
+
+  def copy() = new MaxAggregator[T, BoxedT]()
+}
+
+class MinAggregator[T, BoxedT >: Null](implicit ev: NumericPair[T, BoxedT], ct: ClassTag[T]) extends TypedAggregator[BoxedT] {
+
+  import ev.numeric
+  import Ordering.Implicits._
+
+  var _state: T = ev.numeric.zero
+  var _isNull: Boolean = true
+
+  def result: BoxedT = if (_isNull) null else ev.box(_state)
+
+  def seqOp(x: Any) {
+    if (x != null && (_isNull || ev.unbox(x.asInstanceOf[BoxedT]) < _state)) {
+      _isNull = false
+      _state = ev.unbox(x.asInstanceOf[BoxedT])
+    }
+  }
+
+  def combOp(agg2: this.type) {
+    if (!agg2._isNull && (_isNull || agg2._state < _state))
+      _state = agg2._state
```

Comment:
You need to update _isNull here.  Your tests didn't catch this since they use only a single partition.